### PR TITLE
Add logical and bitwise operators for Java

### LINKED ISSUE #620: Java missing logical operators
# Language

Java (all versions)

# Wrong Field(s)
1. _Logical AND operator_
1. _Logical OR operator_
1. _Logical NOT operator_

These are all supported in Java with syntax almost identical to C-style languages.

### PR DESCRIPTION
**Pull request by jeff303**
## What GitHub issue does this PR apply to?

Resolves #620

## What changed and why?

Added logical operators that are supported in Java.  The "and assignment" variants are not implemented in Java (see [here](https://stackoverflow.com/q/2324549)).

## Checklist

- [X] I claimed any associated issue(s) and they are not someone else's
- [X] I have looked at documentation to ensure I made any revisions correctly
- [X] I tested my changes locally to ensure they work
- [ ] (If editing Django) I have added or edited any appropriate unit tests for my changes

### CODE REVIEW

**Review by gutogalego — APPROVED** [review]
This looks good. There are still a couple of missing bitwise operators, [reference here](https://www.geeksforgeeks.org/bitwise-operators-in-java/), although I don't know when they were introduced.

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]
Like @gutogalego said, there's some missing operators. (I know you made the issue originally for just logic operators but it really makes sense to make the PR update all the operators, not just some of them.)

Also it'd be helpful to show where the operands go relative to operators. (Most languages do like `a + b` but some like Lisp do `(+ a b)` so I'm trying to get those updated.)

If you can add those in, I can merge this in.

image

**Comment by jeff303** [comment]
> Like @gutogalego said, there's some missing operators. (I know you made the issue originally for just logic operators but it really makes sense to make the PR update all the operators, not just some of them.)
> 
> Also it'd be helpful to show where the operands go relative to operators. (Most languages do like `a + b` but some like Lisp do `(+ a b)` so I'm trying to get those updated.)
> 
> If you can add those in, I can merge this in.

I updated the Java 11 language file with all of these suggestions.  Once this looks good, I can apply the same changes to the other Java versions (since all of this will be identical across all three).

File: web/thesauruses/java/15/operators.json [hunk 1]
```diff
@@ -106,6 +106,24 @@
       "not-implemented": "true",
       "name": "Is not operator"
     },
+    "logical_and": {
```
Comment (geekygirlsarah) [inline web/thesauruses/java/15/operators.json:130]: You added operand placement in Java 11's file but not 15 and 17. Could you do the same for those as well?

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]
Make similar changes to v15 and v17 and this should be good to merge!

File: web/thesauruses/java/11/operators.json [hunk 2]
```diff
@@ -31,27 +31,48 @@
...
+      "name": "Unary plus operator"
+    },
+    "unary_minus": {
+      "code": "y = -12",
+      "name": "Unary minus operator"
     },
     "increment": {
-      "code": "++",
+      "comment": "Supports both pre and post increment",
+      "code": "x++\n++x",
```
Comment (geekygirlsarah) [inline web/thesauruses/java/11/operators.json]: Your examples are post-increment then pre-increment, but mention it's pre then post. Can you swap one of those so they match?

File: web/thesauruses/java/11/operators.json [hunk 3]
```diff
@@ -31,27 +31,48 @@
...
       "name": "Increment (add 1) operator"
     },
     "decrement": {
-      "code": "--",
+      "comment": "Supports both pre and post decrement",
+      "code": "x--\n--x",
```
Comment (geekygirlsarah) [inline web/thesauruses/java/11/operators.json]: Same here.

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]
Two tiny swaps and I'll merge it!

**Comment by jeff303** [comment]
> Two tiny swaps and I'll merge it!

Finally got these all fixed up (I believe).  Tested the comparison page between 11-15 and 11-17.

**Review by geekygirlsarah — APPROVED** [review]
Great! I'll merge this in. Thanks for the changes!

**Pull request merged**